tools/tool-testing/selftest.js

```diff
@@ -1856,8 +1868,108 @@ class TestList {
   // Mark a test's file as having failures.  This prevents
   // saveTestState from saving its hash as a potentially
   // "unchanged" file to be skipped in a future run.
-  notifyFailed(test) {
+  notifyFailed(test, failure) {
     this.fileInfo[test.file].hasFailures = true;
+    test.failure = failure || true;
```

Comment:
I think this should be
```js
notifyFailed(test, failure = true) {
  this.fileInfo[test.file].hasFailures = true;
  test.failure = failure;
}
```
If someone calls `testList.notifyFailed(test, false)`, then `test.failure = failure || true` will set `test.failure` to `true`, which seems surprising.